src-java/kilda-model/src/main/java/org/openkilda/model/Cookie.java

```diff
@@ -44,102 +38,133 @@
+            ServiceCookie.ServiceCookieTag.LLDP_POST_INGRESS_VXLAN_COOKIE).getValue();
+    public static final long LLDP_POST_INGRESS_ONE_SWITCH_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.LLDP_POST_INGRESS_ONE_SWITCH_COOKIE).getValue();
+    public static final long ARP_INPUT_PRE_DROP_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.ARP_INPUT_PRE_DROP_COOKIE).getValue();
+    public static final long ARP_TRANSIT_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.ARP_TRANSIT_COOKIE).getValue();
+    public static final long ARP_INGRESS_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.ARP_INGRESS_COOKIE).getValue();
+    public static final long ARP_POST_INGRESS_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.ARP_POST_INGRESS_COOKIE).getValue();
+    public static final long ARP_POST_INGRESS_VXLAN_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.ARP_POST_INGRESS_VXLAN_COOKIE).getValue();
+    public static final long ARP_POST_INGRESS_ONE_SWITCH_COOKIE = new ServiceCookie(
+            ServiceCookie.ServiceCookieTag.ARP_POST_INGRESS_ONE_SWITCH_COOKIE).getValue();
 
-    // 9 bits cookie type "field"
-    public static final long TYPE_MASK                               = 0x1FF0_0000_0000_0000L;
-    public static final long FLOW_COOKIE_TYPE                        = 0x0000_0000_0000_0000L;
-    public static final long LLDP_INPUT_CUSTOMER_TYPE                = 0x0010_0000_0000_0000L;
-    public static final long MULTITABLE_ISL_VLAN_EGRESS_RULES_TYPE   = 0x0020_0000_0000_0000L;
-    public static final long MULTITABLE_ISL_VXLAN_EGRESS_RULES_TYPE  = 0x0030_0000_0000_0000L;
-    public static final long MULTITABLE_ISL_VXLAN_TRANSIT_RULES_TYPE = 0x0040_0000_0000_0000L;
-    public static final long MULTITABLE_INGRESS_RULES_TYPE           = 0x0050_0000_0000_0000L;
-    public static final long ARP_INPUT_CUSTOMER_TYPE                 = 0x0060_0000_0000_0000L;
+    @JsonCreator
+    public Cookie(long value) {
+        super(value);
+    }
 
-    private final long value;
+    @Builder
+    public Cookie(CookieType type) {
+        super(0, type);
+    }
 
     /**
-     * Create {@code Cookie} instance and perform it's validation.
+     * Conver existing {@link Cookie} instance into {@link CookieBuilder}.
      */
-    public static Cookie decode(long rawValue) {
-        Cookie cookie = new Cookie(rawValue);
-        cookie.ensureNoFlagsConflicts();
-        return cookie;
+    public CookieBuilder toBuilder() {
+        return new CookieBuilder()
+                .type(getType());
     }
 
-    @JsonCreator
-    public Cookie(long value) {
-        this.value = value;
-    }
-
-    public static Cookie buildForwardCookie(long unmaskedCookie) {
-        return new Cookie(unmaskedCookie | Cookie.FLOW_PATH_FORWARD_FLAG);
+    protected Cookie(long blank, CookieType type) {
```

Comment:
Why is the value called "blank"?

Reply:
Because it is incomplete yet... Can't find normal name for this argument. If not `blank` only `stub` is coming to the mind. Renamed it into `value` :(